### ISSUE TITLE
[bug] Empty modal after wrong input

### ISSUE THREAD

**Issue by BogDAAAMN** [appwrite/appwrite#229]
**Description**
After wrongfully trying to create a user in the _Users_ menu and getting _Failed to creat user_ alert, the _add user_ modal is not working anymore. It is just empty.

For instance, I try to create a user with a short password. 

![Screenshot_2019-10-23 Appwrite - Users](https://user-images.githubusercontent.com/22895284/67411476-ff28d180-f5bd-11e9-9b42-027678ba72cd.png)

The response of the request is `400 error` and the `Invalid password` message, which is legit. 

```json
{
    "message":"Invalid password: Password must be between 6 and 32 chars and contain ...",
    "code":400,
    "file":"\/usr\/share\/nginx\/html\/vendor\/utopia-php\/framework\/src\/App.php",
    "line":456
}
```

But afterwards, any time I try to press on the **+** button to create a new user, the model is empty.

![Screenshot_2019-10-23 Appwrite - Users(1)](https://user-images.githubusercontent.com/22895284/67411693-4f079880-f5be-11e9-9e8d-4f4689a6172c.png)

**Reproduce**
* Got to *Users* menu
* Press **+** button to add a new user
* Wrongfully add a user
* Get the red alert with `Failed to add user`
* Press again on **+** button to try again

**Desktop**
* OS: MacOS Mojave 10.14
* Browser: Firefox 7.0.2 (64-bit)

**Comment by eldadfux** [comment]
https://github.com/appwrite/appwrite/commit/eb288c5ce2a770ea1c049d5cd56cf936401c146d

This commit should fix it. Will be released in 0.4.0, also available now on master branch.

**Comment by BogDAAAMN** [comment]
ok, perfect! 

Based on this issue I was thinking of a feature that would also show what kind of error is, since we get it from the response. Not only _Failed to create user_ or _Failed to create team_

I can make an issue and also work on it, if you find it useful @eldadfux!

**Comment by eldadfux** [comment]
I think we should treat this as per case thing instead of a simple project-wide task that can affect all errors.

Exposing too much information when throwing errors can be considers as a security weakness in some cases and in other cases just passing to the client the API error AS IS can have a really bad UX as it might be too technical or just too  long.

So basically we have the API error that are thrown from exceptions. Our error handler only show 4** error to the users and all other errors get 5** server error generic message. In dev mode the error are much more verbose for debugging.

The console manages errors in a per form error message logic.

@BogDAAAMN let me know what you think.

**Comment by BogDAAAMN** [comment]
Oh yeah, of course! I wasn't thinking about printing the API errors AS THEY ARE, just getting a friendlier message. 

For instance, when I used a weak password, instead of _Failed to create user_ could've been _Password too weak_ or something equivalent. But you solved this issue in UX with [eb288c5](https://github.com/appwrite/appwrite/commit/eb288c5ce2a770ea1c049d5cd56cf936401c146d), so atm I don't really have another use case. 

I'll see if something else pops up.

**Comment by eldadfux** [comment]
Thats a valid point. I think it will be good to review all the errors thrown inside the controllers and makes sure they are consisted.

Another thing I wanted to improve is to force better error handling inside the Utopia PHP validator interface. Currently this validator only support a description and that's not really the same thing as an error message.
https://github.com/utopia-php/framework/blob/master/src/Validator.php

Thinking about adding a message setter and getter and make Utopia use the message instead of the description when reporting errors.
